### ISSUE TITLE
__items__ is not idiomatic YAML 

### ISSUE THREAD

**Issue by beccasaurus** [googleapis/sample-tester#50]
I find `__items__` to be surprising. I don't undestand why it has `__` but none of the other keys do.

How about `samples`?

```yaml
sets:
- environment: x
  samples:
  - sample: y_z
  - sample: a_b
```

**Comment by vchudnov-g** [comment]
There are two issues here

* Everything under `sets` is a list of arbitrary labels (key/value pairs), meaning that both the keys and values are defined by the user. We need a key in the spirit of `__items__` to separate the common values applied to all items from the list of distinct items (which get all the common values applied). I used dunder around `items` in order to not pollute the key namespace: it is not unreasonable for the user to define an `items` label key.
* The manifest file format is meant to be general, potentially for more than just samples (even though that is the only use case right now). I make a distinction between the manifest file schema semantics vs the semantics of the labels the sample generator provides for the samplegen use case

So, I am willing to use some other convention to designate the grouping list header, instead of `__items__`, though I do consider it important that we have such a convention.

One idea (off the top of  my head, may have issues) is to have a prefix symbol such as `+`. Then any field with the prefix denotes a list of items to which the labels above the list are distributed, regardless of the actual name. The name that follows the `+` has no intrinsic semantic meaning in the manifest file schema, but may have semantic meaning for people or tools reading the file and. Thus, in the sample use case, we can just call it `+samples:` 

In other words
```
sets:
- environment: x
  +samples:
  - name: foo
    id: 57
    items: [alpha, beta]
  - idea: bar
    status: wild
    samples: [easy, hard]
```
is equivalent to:
```
sets:
- +samples:
  - name: foo
    id: 57
    items: [alpha, beta]
    environment: x
  - idea: bar
    status: wild
    samples: [easy, hard]
    environment: x
```

and tools reading this file will know this is a list called `samples`, distinct from any `samples` label in any of the list elements.

**Comment by beccasaurus** [comment]
> Everything under sets is a list of arbitrary labels (key/value pairs), meaning that both the keys and values are defined by the user. We need a key in the spirit of __items__ to separate the common values applied to all items from the list of distinct items (which get all the common values applied). 

Why does there need to be something special? I've never seen something like `+something` in a YAML key.

I'm missing something :) What's wrong with this?

```yaml
version: 1
sets:
- environment: php
  bin: php
  path: talent/v4beta1/
  any_regular_key_can_go_here:
  - sample: job_search_create_company
    path: php/JobSearchCreateCompany.php
```

```proto
message SampleSet {
  string environment = 1;
  string bin = 2;
  string path = 3;
  repeated Map<string,string> samples;
}
```

**Comment by vchudnov-g** [comment]
Oooh! I now think there's **a better way** to do this with YAML anchors ([ref](https://blog.daemonl.com/2016/02/yaml.html)). This will achieve everything I wanted to do, is standard, and is simpler to implement. I'll flesh this out shortly, but it will be _standard_! Yay!

The answer I was typing (before I discovered this other approach) follows for clarity as to my previous thinking, but now I advocate using anchors instead.

"""
_In your example, we're removing the label `any_regular_key_can_go_here` from the namespace of possible labels for each item._

_The manifest file itself is meant to be just a listing of items, each with an arbitrary set of key/value pairs. The purpose of having the `__items__` was to avoid duplicating key/value pairs that are shared among all the items, by allowing them to be factored out above the `__items__` list. Whatever name we give to the `__items__` list, that name can no longer be factored out above the list._
"""

**Comment by vchudnov-g** [comment]
New proposed schema using YAML anchors: http://go/actools-samplegen-manifest-v3 (sorry, Google-internal link only). Thoughts?

**Comment by vchudnov-g** [comment]
Addressed in manifest v3 (#74)
